code/game/turfs/closed/walls.dm

```diff
@@ -34,48 +34,63 @@
 
 	var/list/dent_decals
 
-/turf/closed/wall/MouseDrop_T(mob/living/carbon/carbon_mob, mob/user)
+/turf/closed/wall/MouseDrop_T(atom/dropping, mob/user, params)
 	..()
-	if(carbon_mob != user)
+	if(dropping != user)
 		return
-	if(carbon_mob.is_leaning == TRUE)
+	if(!iscarbon(dropping) && !iscyborg(dropping))
 		return
-	if(carbon_mob.pulledby)
+	var/mob/living/leaner = dropping
+	if(leaner.incapacitated(IGNORE_RESTRAINTS) || leaner.stat != CONSCIOUS || HAS_TRAIT(leaner, TRAIT_NO_TRANSFORM))
 		return
-	if(!carbon_mob.density)
+	if(!leaner.density || leaner.pulledby || leaner.buckled || !(leaner.mobility_flags & MOBILITY_STAND))
```

Comment:
we can couple the has_trait with `density` since they both will exist on the `/atom` sublevel. like

```dm
if(!dropping.density || HAS_TRAIT(dropping, TRAIT_NO_TRANSFORM))
 return
```

granted the whole thing about it being `dropping` might not read as well but i figured i'd just mention it here now rather than ponder about it later

Reply:
I don't really see the need for this, it's not like the cast is very hot and it just makes it less clear to read